eclair-core/src/main/scala/fr/acinq/eclair/channel/Helpers.scala

```diff
@@ -1078,7 +1078,11 @@ object Helpers {
         log.error(s"some htlcs don't have a corresponding timeout transaction: tx=$tx, htlcs=$matchingHtlcs, timeout-txs=$matchingTxs")
       }
       matchingHtlcs.zip(matchingTxs).collectFirst {
+        // HTLC transactions cannot change when anchor outputs is unused, so we directly check the txid
         case (add, timeoutTx) if timeoutTx.txid == tx.txid => add
+        // Claim-HTLC transactions can be updated to pay more or less fees by changing the output amount, so we cannot
+        // rely on txid equality: we instead check that the input is the same and the output goes to the same address.
+        case (add, timeoutTx) if timeoutTx.txIn.head.outPoint == tx.txIn.head.outPoint && timeoutTx.txOut.head.publicKeyScript == tx.txOut.head.publicKeyScript => add
```

Comment:
See https://github.com/ACINQ/eclair/pull/2113#discussion_r782826106, if you have an idea on how to improve this without adding too much extra code, that would be great!

Reply:
I would simply remove the second equality, but I'm certainly missing something:

```suggestion
        // Claim-HTLC transactions can be updated to pay more or less fees by changing the output amount, so we cannot
        // rely on txid equality: we instead check that the input is the same and the output goes to the same address.
        case (add, timeoutTx) if timeoutTx.txIn.head.outPoint == tx.txIn.head.outPoint => add
```

Reply:
If you remove this, there is a risk you match on the remote `claim-htlc-success` instead of your `htlc-timeout` (they spend the same input, but to different addresses), that's why I think we need both.

Reply:
No, because their witness script is different and `claim-htlc-success` would be filtered out (see `Scripts.extractPaymentHashFromHtlcTimeout` and `Scripts.extractPaymentHashFromClaimHtlcTimeout`).

Reply:
The issue with the current belt and braces is that it makes the intent less clear and indeed I got confused. Not sure that's a plus on the safety side?

I understand the need for redundancy though, especially since we don't type-check the method arguments, but how about adding a `require` at the beginning of the method, maybe on the `witness` length or something?

Back to my initial [comment](https://github.com/ACINQ/eclair/pull/2113#discussion_r782346371), now that we agree on the first part, what about the second part?

> Also, if we are confident with the matching case for the anchor outputs case, why don't we only rely on this one?

Reply:
Done in https://github.com/ACINQ/eclair/pull/2113/commits/29b5039bd3c99b8a87ce762ab49db382cf24d962